diffusion/learned.py

```diff
@@ -10,47 +10,104 @@
 class LearnedGaussianDiffusion(GaussianDiffusion):
     """Implements the core learning and inference algorithms."""
     def __init__(
-        self, noise_model, z_encoder_model, timesteps, img_shape,
+        self, noise_model, forward_matrix, timesteps, img_shape,
         schedule='linear', device='cpu'
     ):
         super().__init__(
             noise_model, timesteps, img_shape, schedule, device
         )
-        self.z_encoder_model = z_encoder_model
+        self.forward_matrix = forward_matrix
         self.z_shape = img_shape
 
-    def z_sample(self, x0, noise=None):
-        """Samples an auxiliary latent a."""
-        if noise is None: 
-            noise_shape = [x0.shape[0]] + self.z_shape
-            noise = torch.randn(noise_shape, device=self.device)
-
-        mu_z, logsigma_z = self.z_encoder_model(x0)
-        z = mu_z + torch.exp(logsigma_z) * noise
-
-        return z, mu_z, logsigma_z
+    def _forward_sample(self, x0, time):
+        return self.forward_matrix(self.model.time_mlp(time))
 
     def q_sample(self, x0, t, noise=None):
         """Samples from the forward diffusion process q.
 
-        Takes a sample from q(xt|x0).
+        Takes a sample from q(xt|x0). t \in {1, \dots, timesteps}
         """
         # encode x0 into auxiliary encoder variable a
-        if noise is None:  noise, _, _ = self.z_sample(x0)
-        return super().q_sample(x0, t, noise)
+        if noise is None:  noise, _, _ = torch.randn_like(x0)
```

Comment:
Also, how does this PR integrate with your previous PR? Aren't you proposing two different sets of modifications to the same function?